File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/entities/TenantIpAllowlistEntity.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.entities;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "tenant_ip_allowlist")
@SQLDelete(sql = "UPDATE identity.tenant_ip_allowlist SET deleted_at = NOW() AT TIME ZONE 'UTC' WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
public class TenantIpAllowlistEntity {

    @Id
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(name = "cidr", nullable = false)
    private String cidr;

    @Column(name = "description")
    private String description;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;

    protected TenantIpAllowlistEntity() {
    }

    public TenantIpAllowlistEntity(
        UUID id,
        UUID tenantId,
        String cidr,
        String description,
        Instant createdAt,
        Instant deletedAt
    ) {
        this.id = id;
        this.tenantId = tenantId;
        this.cidr = cidr;
        this.description = description;
        this.createdAt = createdAt;
        this.deletedAt = deletedAt;
    }

    public UUID getId() {
        return id;
    }

    public UUID getTenantId() {
        return tenantId;
    }

    public String getCidr() {
        return cidr;
    }

    public String getDescription() {
        return description;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getDeletedAt() {
        return deletedAt;
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/entities/TenantIpAllowlistEntity.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.entities

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.SQLDelete
import org.hibernate.annotations.SQLRestriction
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "tenant_ip_allowlist")
@SQLDelete(sql = "UPDATE identity.tenant_ip_allowlist SET deleted_at = NOW() AT TIME ZONE 'UTC' WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
class TenantIpAllowlistEntity (

    @Id
    var id: UUID,

    @Column(name = "tenant_id", nullable = false)
    var tenantId: UUID,

    @Column(name = "cidr", nullable = false)
    var cidr: String,

    @Column(name = "description")
    var description: String?,

    @Column(name = "created_at")
    var createdAt: Instant = Instant.now(),

    @Column(name = "deleted_at")
    var deletedAt: Instant? = null
){}